index.html

```diff
@@ -3750,6 +3777,180 @@ <h2>
+          <p data-test=
+          "payment-request/MerchantValidationEvent/complete-method-manual.https.html">
+            The <a>MerchantValidationEvent</a>'s
+            <code>complete(<var>merchantSessionPromise</var>)</code> method
+            MUST act as follows:
+          </p>
+          <ol class="algorithm">
+            <li>Let <var>event</var> be the <a>context object</a>.
+            </li>
+            <li>If <var>event</var>'s <a>isTrusted</a> attribute is false, then
+            then <a>throw</a> an "<a>InvalidStateError</a>"
+            <a>DOMException</a>.
+            </li>
+            <li>Let <var>request</var> be <var>event</var>.<a data-lt=
+            "MVE.[[request]]">[[\request]]</a>.
+            </li>
+            <li>If <var>request</var>.<a>[[\state]]</a> is not
+            "<a>interactive</a>", then <a>throw</a> an
+            "<a>InvalidStateError</a>" <a>DOMException</a>.
+            </li>
+            <li>If <var>request</var>.<a>[[\updating]]</a> is true, then
+            <a>throw</a> an "<a>InvalidStateError</a>" <a>DOMException</a>.
+            </li>
+            <li>Set <var>event</var>'s <a>stop propagation flag</a> and <a>stop
+            immediate propagation flag</a>.
+            </li>
+            <li>Set <var>event</var>.<a>[[\waitForUpdate]]</a> to true.
+            </li>
+            <li>Run the <a>validate a merchant algorithm</a> with
+            <var>merchantSessionPromise</var> and <var>request</var>.
+            </li>
+          </ol>
+        </section>
+        <section data-dfn-for="MerchantValidationEventInit" data-link-for=
+        "MerchantValidationEventInit">
+          <h3>
+            <dfn>MerchantValidationEventInit</dfn> dictionary
+          </h3>
+          <pre class="idl">
+            dictionary MerchantValidationEventInit {
+              USVString validationURL = "";
+            };
+          </pre>
+          <section>
+            <h4>
+              <dfn>validationURL</dfn> member
+            </h4>
+            <p>
+              A URL from which a developer would fetch <a>payment
+              handler</a>-specific verification data.
+            </p>
+          </section>
+        </section>
+        <section>
+          <h3>
+            Internal slots
+          </h3>
+          <p>
+            Instances of <a>MerchantValidationEvent</a> are created with the
+            internal slots in the following table:
+          </p>
+          <table>
+            <tr>
+              <th>
+                Internal Slot
+              </th>
+              <th>
+                Description (<em>non-normative</em>)
+              </th>
+            </tr>
+            <tr>
+              <td>
+                <dfn data-lt="MVE.[[request]]" data-lt-nodefault=
+                "">[[\request]]</dfn>
+              </td>
+              <td>
+                The <a>PaymentRequest</a> instance that instantiated this
+                <a>PaymentResponse</a>.
```

Comment:
"this" is not a PaymentResponse, so this doesn't seem right...

Reply:
These slots are not needed anymore. I can get request from `target`, after `isTrusted` is checked.